### DISCUSSION TITLE
Simplest way to override the icon of ShowDevTools

### DISCUSSION THREAD

**Discussion by oscar-articulate** [cefsharp/CefSharp#5014] [Q&A]
In `124.3.80` calling ShowDevTools created a form with the following icon:
![image](https://github.com/user-attachments/assets/4ff5d64b-70ba-4a9f-82a8-0021c2f47533)

In `130.1.90` calling ShowDevTools creates a form with a new icon: 
![image](https://github.com/user-attachments/assets/ba18dcb6-0184-4cc4-9aed-6b2ce14eaa52)

I would like to override this icon with our application's icon.

Currently what I'm doing is:

```
var devToolsForm = new Form { Icon = Properties.Resources.Icon }; 
var devToolsControl = CefSharp.WinForms.WebBrowserExtensions.ShowDevToolsDocked(this, devToolsForm);
devToolsForm.Show(mainApplicationForm);

```

This is fine but it also requires a lot of other code to handle keyboard, resizing, etc.

Is there a simpler way to do this?

**Comment by amaitland** [comment]
It looks like `CEF` has an option for setting the icon, only when using the `Chrome Styling` (which isn't the default in `CefSharp`).

It's not currently exposed as part of the `API`.

https://cef-builds.spotifycdn.com/docs/131.3/structcef__settings__t.html#ae35fa69c27617b2b6c40f083aa614789

Pull request welcome.

**Reply by amaitland** [reply]
I cannot remember if you can get the `HWND` for the `DevTools` window, If you can, then you can probably use https://learn.microsoft.com/en-us/windows/win32/winmsg/wm-seticon
